### DISCUSSION TITLE
How to troubleshoot hydrating errors?

### DISCUSSION THREAD

**Discussion by ric980** [remix-run/remix#5648] [Q&A] [answered]
Hello,

My app is spitting hydration errors every second. I have 2.5K errors per hour. These are the error types:

- There was an error while hydrating. Because the error happened outside of a Suspense boundary, the entire root will switch to client rendering.
- Text content does not match server-rendered HTML.
- Hydration failed because the initial UI does not match what was rendered on the server.

So, my app is running client side without any SSR.

The only thing I can think of that could be causing the problem is that in the loader the fetch of resources never returns the same data twice. Its a highly concurrent application that aggregates tons of data and I cannot guarantee that the same GET request returns the same JSON response. Every server response is different.

I am using no Chrome extensions.
I don't have any divs inside paragraphs.
I already checked StackOverflow for possible solutions, but nothing seems to work.
The app is private so I cannot share source code.

My questions are:
- How can I troubleshoot what's wrong? These errors give me no hint.
- How to disable this client vs server comparison?

Anyway, at this point I am clueless and thinking about switching the framework before it's too late, i.e. the app gets bigger and the problems gets bigger.

**Comment by lili21** [comment]
> I have 2.5K errors per hour

I guess your users are visit your app, and maybe they are using Chrome extension.

> How can I troubleshoot what's wrong? These errors give me no hint.

can you reproduce the error in local dev? Then maybe you can get more detailed error message.

> How to disable this client vs server comparison?

that's the process of hydration. You can remove the `<Scripts />` to disable it, but there will be no interactive.

**Reply by AnonSols** [reply]
I newly installed  a next js application on my pc, after that I ran the command using using ' npm run dev', it compiled then showed the next js landing page while the page was loading. After the page was done loading I got an application error, I checked my console and I saw this **'hyrating errors**'. Please I really need this fixed how can I go around this? I have tried solutions online but none seems to be working.

And not just even creating a new next app, even older doesn't compile properly anymore, I have updated my chrome and done other things I saw from the internet. 

Really appreciate a response thanks

**Reply by lili21** [reply]
Try this , https://github.com/BuilderIO/hydration-overlay

**Comment by sergiodxa** [comment, marked as answer]
> So, my app is running client side without any SSR

This is not what the error means, the error means that instead of hydrating the HTML generated during SSR what React will do is to render the app and throw away the HTML, here the difference is that hydrating just run effects and attach event listeners, while the other process is generating the DOM again.

You still have SSR, but it will take a few ms more to finish the hydration/rendering process.

> The only thing I can think of that could be causing the problem is that in the loader the fetch of resources never returns the same data twice. Its a highly concurrent application that aggregates tons of data and I cannot guarantee that the same GET request returns the same JSON response. Every server response is different.

This can't happen, because when React hydrates the app client-side the loaders are not being fetched again, instead Remix sends the data it got from the loader during SSR in the HTML and use it during the hydration, so it's always the same data.

What could be happening is dates being parsed with a different timezone during SSR and CSR for example.

**Reply by ric980** [reply]
Thank you so much @sergiodxa. That was it. I'm using `toLocaleDateString`, but the locale on the server and the client is different, so it produces different date string. I've changed the server locale and the number of errors decreased significantly. Now I get only a few errors a day, I guess from people using browser extensions. Thanks again. Back to loving Remix again 🎉

**Reply by lili21** [reply]
@ric980 I would recommend you to do it with [ClientOnly](https://github.com/sergiodxa/remix-utils#clientonly) instead of change server locale.
or you can try this [suppressHydrationWarning](https://beta.reactjs.org/reference/react-dom/hydrate#suppressing-unavoidable-hydration-mismatch-errors)

**Reply by sergiodxa** [reply]
You could also parse dates in the loader so the timezone is used once there.

**Comment by Badru9** [comment]
and its possible that this error caused by third-party framework
